### PR TITLE
add default dependenciesInterpreter framework

### PR DESCRIPTION
**Pull request by mrlihanbo**
Signed-off-by: lihanbo <lihanbo2@huawei.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
add default dependenciesInterpreter framework

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/resourceinterpreter/defaultinterpreter/default.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ type DefaultInterpreter struct {
 	reviseReplicaHandlers   map[schema.GroupVersionKind]reviseReplicaInterpreter
 	retentionHandlers       map[schema.GroupVersionKind]retentionInterpreter
 	aggregateStatusHandlers map[schema.GroupVersionKind]aggregateStatusInterpreter
+	dependenciesHandler     map[schema.GroupVersionKind]dependenciesInterpreter
```
Comment (iawia002) [inline pkg/resourceinterpreter/defaultinterpreter/default.go:21]:
```suggestion
	dependenciesHandlers    map[schema.GroupVersionKind]dependenciesInterpreter
```

**Review by iawia002 — COMMENTED** [review]
Basically LGTM, just one comment

File: pkg/resourceinterpreter/defaultinterpreter/default.go (hunk 1 above)
Comment (RainbowMango) [reply pkg/resourceinterpreter/defaultinterpreter/default.go]: Nice catch!!
+1

Comment (mrlihanbo) [reply pkg/resourceinterpreter/defaultinterpreter/default.go]: > 
Done! Thanks for your review.

File: pkg/resourceinterpreter/interpreter.go [hunk 2]
```diff
@@ -153,3 +155,22 @@ func (i *customResourceInterpreterImpl) AggregateStatus(object *unstructured.Uns
 
 	return i.defaultInterpreter.AggregateStatus(object, aggregatedStatusItems)
 }
+
+// GetDependencies returns the dependencies' info of the given object.
```
Comment (RainbowMango) [inline pkg/resourceinterpreter/interpreter.go:159]: comments.

File: pkg/resourceinterpreter/defaultinterpreter/default.go [hunk 3]
```diff
@@ -92,3 +97,12 @@ func (e *DefaultInterpreter) AggregateStatus(object *unstructured.Unstructured,
 	}
 	return handler(object, aggregatedStatusItems)
 }
+
+// GetDependencies returns the dependencies' info of the given object.
```
Comment (RainbowMango) [inline pkg/resourceinterpreter/defaultinterpreter/default.go:101]: comments...

**Comment by pigletfly** [comment]
do we want to support customized depend interperter? For example, we have a crd's cr which will create another crd's cr.
```
apiVersion: xxx
kind: XXX
metadata:
  name: abc
spec:
  disk:
     - size: 10
```
CRD XXX has a disk in spec, which is another CRD that should be created when XXX is created.

**Comment by mrlihanbo** [comment]
> do we want to support customized depend interperter? For example, we have a crd's cr which will create another crd's cr.
> 
> ```
> apiVersion: xxx
> kind: XXX
> metadata:
>   name: abc
> spec:
>   disk:
>      - size: 10
> ```
> 
> CRD XXX has a disk in spec, which is another CRD that should be created when XXX is created.

For now, dependenciesInterpreter will be used in `dependncies distributor`. CRD XXX has a disk in spec, if XXX is propagated to cluster A, B, C, the disk will be automatically propagated to cluster A, B, C.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/lgtm

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**